reducers/views/channel_selector_modal.ts

```diff
@@ -3,12 +3,14 @@
 
 import {combineReducers} from 'redux';
 
+import type {GenericAction} from 'mattermost-redux/types/actions';
+
 import {ChannelTypes, UserTypes} from 'mattermost-redux/action_types';
 
-function channels(state = [], action) {
+function channels(state: string[] = [], action: GenericAction) {
     switch (action.type) {
     case ChannelTypes.RECEIVED_ALL_CHANNELS:
-        return action.data.map((v) => v.id);
+        return action.data.map((v: any) => v.id);
```

Comment:
Wouldn't `v` be of type `Channel` here?
```suggestion
        return action.data.map((v: Channel) => v.id);
```